runChecks.py

```diff
@@ -0,0 +1,134 @@
+    return errors
+
+def check_non_test_files_in_test():
+    all_cpp = files_in_folder("test/unit/*/*")
+    # if the file is a .cpp file that doesnt end with _test.cpp
+    errors = ["Error: A .cpp file without the suffix " + testsfx + " in test/unit/*/*:\n\t" + x for x in all_cpp if x[-4:] == ".cpp" and x[-len(testsfx):] != testsfx]
+    return errors
+
+def main():
+    errors = []
+
+    # Check for files inside stan/math/prim that contain stan/math/rev or stan/math/fwd
+    prim_checks = [
+        { "pattern" : '<stan/math/rev/', "message" : 'File includes a stan/math/rev header file.'},
+        { "pattern" : 'stan::math::var', "message" : 'File uses stan::math::var.'},
+        { "pattern" : '<stan/math/fwd/', "message" : 'File includes a stan/math/fwd header file.'},
+        { "pattern" : 'stan::math::fvar', "message" : 'File uses stan::math::fvar.'},
+        { "pattern" : '<stan/math/mix/', "message" : 'File includes a stan/math/mix header file.'}
+    ]
+    errors += grep_patterns('prim', 'stan/math/prim', prim_checks)
+    
+    # Check for files inside stan/math/rev that contain stan/math/fwd or stan/math/mix
+    rev_checks = [
+        { "pattern" : '<stan/math/fwd/', "message" : 'File includes a stan/math/fwd header file.'},
+        { "pattern" : 'stan::math::fvar', "message" : 'File uses stan::math::fvar.'},
+        { "pattern" : '<stan/math/mix/', "message" : 'File includes a stan/math/mix header file.'}
+    ]
+    errors += grep_patterns('rev', 'stan/math/rev', rev_checks)
+
+    # Check for files inside stan/math/*/scal that contain stan/math/*/arr or stan/math/*/mat
+    scal_checks = [
+        { "pattern" : '<stan/math/.*/arr/', "message" : 'File includes an array header file.'},
+        { "pattern" : '<vector>', "message" : 'File includes an std::vector header.'},
+        { "pattern" : 'std::vector', "message" : 'File uses std::vector.'}, 
+        { "pattern" : '<stan/math/.*/mat/', "message" : 'File includes a matrix header file.'}, 
+        { "pattern" : '<Eigen', "message" : 'File includes an Eigen header.'},
+        { "pattern" : 'Eigen::', "message" : 'File uses Eigen.'}
+    ]
+    errors += grep_patterns('scal', 'stan/math/*/scal', scal_checks)
+
+    # Check for files inside stan/math/*/arr that contain stan/math/*/mat or Eigen
+    arr_checks = [
+        { "pattern" : '<stan/math/.*/mat/', "message" : 'File includes an matrix header file.'},
+        { "pattern" : '<Eigen', "message" : 'File includes an Eigen header.'},
+        { "pattern" : 'Eigen::', "message" : 'File uses Eigen.'}
+    ]    
+    errors += grep_patterns('arr', 'stan/math/*/arr', arr_checks)
+
+    # Check to make sure we use C++14 constructs in stan/math
+    cpp14_checks = [
+        { "pattern" : 'boost::is_unsigned<', "message" : 'File uses boost::is_unsigned instead of std::is_unsigned.'},
+        { "pattern" : '<boost/type_traits/is_unsigned>',\
```

Comment:
`\` is not needed if the line continues with the same indentation.

```
[
    {"pattern" :  "patt",
     "message" : "patt2"},
```

Reply:
Thanks!